spinoso-time/src/time/tzrs/error.rs

```diff
@@ -0,0 +1,205 @@
+            // Occurs when calling system clock
+            TzError::SystemTimeError(_) => Self::UnknownTzError(error),
+            // Occurs during parsing of TZif files
+            TzError::TzFileError(_) => Self::UnknownTzError(error),
+            // Occurs during parsing of TZif files (POSIX string parsing)
+            TzError::TzStringError(_) => Self::UnknownTzError(error),
+            // Occurs during int conversion (e.g. i64 => i32)
+            TzError::OutOfRangeError(_) => Self::UnknownTzError(error),
+            // Occurs during creation of TimeZoneRef
+            TzError::LocalTimeTypeError(_) => Self::UnknownTzError(error),
+            // Occurs during creation of TimeZoneRef
+            TzError::TransitionRuleError(_) => Self::UnknownTzError(error),
+            // Occurs during creation of TimeZoneRef
+            TzError::TimeZoneError(_) => Self::UnknownTzError(error),
+            // Wrapped by ProjectDateTimeError
+            TzError::FindLocalTimeTypeError(_) => Self::UnknownTzError(error),
+            // Never explicitly returned
+            TzError::IoError(_) => Self::UnknownTzError(error),
+            // Never explicity returned
+            TzError::Utf8Error(_) => Self::UnknownTzError(error),
+            // Never explicitly returned
+            TzError::TryFromSliceError(_) => Self::UnknownTzError(error),
+            // TzError is non_exhaustive, so the rest are caught as Unknown
+            _ => Self::Unknown,
+        }
+    }
+}
+
+impl From<TzStringError> for TimeError {
+    fn from(err: TzStringError) -> Self {
+        Self::TzStringError(err)
+    }
+}
+
+impl From<TzOutOfRangeError> for TimeError {
+    fn from(err: TzOutOfRangeError) -> Self {
+        Self::TzOutOfRangeError(err)
+    }
+}
+
+impl From<IntOverflowError> for TimeError {
+    fn from(error: IntOverflowError) -> Self {
+        Self::IntOverflowError(error)
+    }
+}
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct TzStringError;
+
+impl fmt::Display for TzStringError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(
+            f,
+            "+HH:MM\", \"-HH:MM\", \"UTC\" or \"A\"..\"I\",\"K\"..\"Z\" expected for utc_offset"
+        )
+    }
+}
+impl error::Error for TzStringError {}
+
+impl From<Utf8Error> for TzStringError {
+    fn from(_: Utf8Error) -> Self {
+        TzStringError
+    }
+}
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct TzOutOfRangeError;
+
+impl fmt::Display for TzOutOfRangeError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(f, "utc_offset out of range")
+    }
+}
+
+impl error::Error for TzOutOfRangeError {}
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct IntOverflowError;
```

Comment:
ah yes. I assume this is because you don't want someone else constructing a `spinoso-securerandom::RandomBytesError` from some completely unrelated crate. Makes sense.

I ran across thiswhen creating the IntOverflowError because the std crate uses this: https://doc.rust-lang.org/stable/src/core/num/error.rs.html#6-9 instead - I probably should have clued into it then too hah.

I'll make the change to keep it consistent with `spinoso` style :+1:

Reply:
Okie dokie, all done in 985d0727.

From my understanding, these errors were still hidden inside the crate internals, and coudn't be constructed outside of the crate. Specifically because the lack of `pub use` on the internal types (only `Timerror` has `pub` from within `mod`). I would claim this was by design, but it was a total fluke. I've added the docs to keep it consistent though.

Does this feel in line with the rest of `spinoso` etc?

Reply:
this LGTM. With `TimeError` we can delegate to `fmt::Display` impl to get the underlying message. The `fn message` is less useful if we need to allocate a `String`.